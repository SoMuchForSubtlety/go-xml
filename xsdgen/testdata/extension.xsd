<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ns="http://example.org"
  targetNamespace="http://example.org">
  <xs:import namespace="http://example2.org" />
  <xs:element name="TypeA">
    <xs:complexType>
      <xs:complexContent>
        <xs:extension base="ns:Response">
          <xs:sequence>
            <xs:element name="intField">
              <xs:simpleType>
                <xs:restriction base="xs:integer">
                  <xs:totalDigits value="15" />
                </xs:restriction>
              </xs:simpleType>
            </xs:element>
          </xs:sequence>
        </xs:extension>
      </xs:complexContent>
    </xs:complexType>
  </xs:element>

  <xs:element name="TypeB">
    <xs:complexType>
      <xs:complexContent>
        <xs:extension base="ns:Response">
          <xs:sequence>
            <xs:element name="stringField">
              <xs:simpleType>
                <xs:restriction base="xs:string">
                </xs:restriction>
              </xs:simpleType>
            </xs:element>
          </xs:sequence>
        </xs:extension>
      </xs:complexContent>
    </xs:complexType>
  </xs:element>

  <xs:complexType name="Response">
    <xs:sequence>
      <xs:element maxOccurs="1" minOccurs="0" name="Error" nillable="true" type="ns:Error">
      </xs:element>
      <xs:element maxOccurs="10" minOccurs="0" name="Warning" nillable="true" type="xs:string">
      </xs:element>
    </xs:sequence>
  </xs:complexType>


  <xs:complexType name="Error">
    <xs:sequence>
      <xs:element maxOccurs="1" minOccurs="0" name="message" nillable="true" type="xs:string">
      </xs:element>
      <xs:element maxOccurs="1" minOccurs="1" name="code" nillable="true" type="xs:string">
      </xs:element>
    </xs:sequence>
  </xs:complexType>
</xs:schema>
