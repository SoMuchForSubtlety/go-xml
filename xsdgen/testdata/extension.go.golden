// Code generated by xsdgen.test. DO NOT EDIT.

package ws

type Error struct {
	Message string `xml:"http://example.org message,omitempty"`
	Code    string `xml:"http://example.org code,omitempty"`
}

type Response struct {
	Error   *Error   `xml:"http://example.org Error,omitempty"`
	Warning []string `xml:"http://example.org Warning,omitempty"`
}

type TypeA struct {
	Error    *Error   `xml:"http://example.org Error,omitempty"`
	Warning  []string `xml:"http://example.org Warning,omitempty"`
	IntField int      `xml:"http://example.org intField"`
}

func (t TypeA) GetError() *Error {
	return t.Error
}
func (t TypeA) GetWarning() []string {
	return t.Warning
}

type TypeB struct {
	Error       *Error   `xml:"http://example.org Error,omitempty"`
	Warning     []string `xml:"http://example.org Warning,omitempty"`
	StringField string   `xml:"http://example.org stringField"`
}

func (t TypeB) GetError() *Error {
	return t.Error
}
func (t TypeB) GetWarning() []string {
	return t.Warning
}
